pyanaconda/modules/network/network.py

```diff
@@ -134,8 +137,16 @@ def process_kickstart(self, data):
             log.debug("updated activate value of the first network command (None -> True)")
 
         self._original_network_data = data.network.network
-        if data.network.hostname:
-            self.set_hostname(data.network.hostname)
+        hostname = data.network.hostname
+        log.debug("Validating hostname: %s", hostname)
```

Comment:
The validation should be done in the `parse` method of the `network` kickstart command. Then you can issue `KickstartParseWarning` and drop the support for `KickstartProcessingWarning`.

Reply:
That's the same argument as invalid UID and GIDs, and that was rejected...

Reply:
Invalid UID or GID doesn't cause a crash of the installation, does it?

Reply:
The place to do this is actually https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/modules/network/kickstart.py#L26

Reply:
Completely changed to use the suggested place instead.